documentation/src/main/wiki/SnippetProcessing.md

```diff
@@ -0,0 +1,194 @@
+#Snippet processing
+##Overview
+In order to fetch the data for snippet different services can be called. Decision which services to call is made depends on service path data attribute and incoming request method.
+
+###GET services calls
+When service path is marked as data-uri-**get**, call will be executed only if http method of incoming request is GET.  
+```html
+<script data-api-type="templating" type="text/x-handlebars-template" 
+    data-uri-get="/service/sample">
+        <h1>Welcome</h1>
+        <h2>{{welcomeMessage}}</h2>
+</script>
+```
+
+###POST services calls
+When service path is marked as data-uri-**post**, call will be executed only if http method of incoming request is POST.
+```html
+<script data-api-type="templating" type="text/x-handlebars-template" 
+    data-uri-post="/service/formSubmit">
+  <p>Please subscribe to the newsletter:</p>
+  <form method="post">
+    <input type="email" name="email" />
+    <input type="submit" value="Submit" />
+  </form>
+  <p>{{message}}</p>
+</script>
+```
+
+###Any method services calls
+When service path is marked as data-uri-**all**, GET call will be execute regardless of incoming request method type.
```

Comment:
Maybe we should mention, that calling `data-uri` without any postfix is alias for `data-uri-all`?